flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorSubmissionTest.java

```diff
@@ -774,6 +687,51 @@ public void testRequestStackTraceSample() throws Exception {
 		}
 	}
 
+	private TaskDeploymentDescriptor createSender(NettyShuffleDescriptor sdd) throws IOException {
+		return createSender(sdd, TestingAbstractInvokables.Sender.class);
+	}
+
+	private TaskDeploymentDescriptor createSender(
+			NettyShuffleDescriptor shuffleDeploymentDescriptor,
+			Class<? extends AbstractInvokable> abstractInvokable) throws IOException {
+		PartitionDescriptor partitionDescriptor = new PartitionDescriptor(
+			new IntermediateDataSetID(),
+			shuffleDeploymentDescriptor.getResultPartitionID().getPartitionId(),
+			ResultPartitionType.PIPELINED,
+			1,
+			0);
+		ResultPartitionDeploymentDescriptor resultPartitionDeploymentDescriptor = new ResultPartitionDeploymentDescriptor(
+			partitionDescriptor,
+			shuffleDeploymentDescriptor,
+			1,
+			true);
+		return createTestTaskDeploymentDescriptor(
+			"Sender",
+			shuffleDeploymentDescriptor.getResultPartitionID().getProducerId(),
+			abstractInvokable,
+			1,
+			Collections.singletonList(resultPartitionDeploymentDescriptor),
+			Collections.emptyList());
+	}
+
+	private TaskDeploymentDescriptor createReceiver(
+			NettyShuffleDescriptor shuffleDescriptor,
+			ResourceID location) throws IOException {
```

Comment:
I guess `location` is not needed in the parameter which could be created directly inside.

Reply:
some tests rely on it to be the same as in some other components outside of this method.